pgml-extension/src/bindings/transformers/transformers.py

```diff
@@ -41,9 +42,21 @@
     PegasusTokenizer,
     TrainingArguments,
     Trainer,
-    GPTQConfig
+    GPTQConfig,
+    TrainerCallback,
 )
 import threading
+import logging
+import evaluate
+import torch.nn.functional as F
+from trl import SFTTrainer, DataCollatorForCompletionOnlyLM
+from trl.trainer import ConstantLengthDataset
+from peft import LoraConfig, get_peft_model
+from pypgrx import print_info, insert_logs
```

Comment:
Need to make sure we either import this conditionally (only for fine tuning) and we include this in requirements.linux.txt. I didn't see a Mac OS build for this and for the M1/M2 architecture, we've been doing releases manually from our Macs (Github actions doesn't have M1 builders). 

This makes me thing we should start cross-compiling soon. Rust supports this pretty well, maturin may need a patch.

Reply:
I couldn't get fine tuning to work on Mac OS. It keeps crashing. How about I check for the operating system and bail out if it is mac? 
requirements.linux.txt is updated with trl and peft.